pkg/controllers/provisioning/provisioner.go

```diff
@@ -100,7 +100,7 @@ func (p *Provisioner) provision(ctx context.Context) error {
 		}
 	}
 	// Separate pods by scheduling constraints
-	schedules, err := p.scheduler.Solve(ctx, p.Provisioner, pods)
+	schedules, err := p.scheduler.Solve(ctx, p.Provisioner, p.cloudProvider, pods)
```

Comment:
Let's inject the cloud provider using dependency injection.

Reply:
done.